indolaw-parser/parser_utils.py

```diff
@@ -562,9 +589,36 @@ def convert_tree_to_json(node: Union[ComplexNode, PrimitiveNode]) -> Dict[str, A
         return {
             'type': node.type.value,
             'id': get_id(node),
-            'children': [convert_tree_to_json(child) for child in node.children],
+            'children': [convert_tree_to_json(child, ketentuan_umum_list) for child in node.children],
         }
 
+def get_parent_node(node: Union[ComplexNode, PrimitiveNode], structure: Structure):
+    if node.parent:
+        if node.parent.type == structure:
+            return node.parent
+        
+        return get_parent_node(node.parent, structure)
+
+    return None
+
+def is_word_part_of_text(string: str, substring: str, start_index) -> bool:
+    '''
+    Determine if a given substring is an independent word that's part of a text.
+    Independent word means that the word is not a substring of another word
+
+    This function checks whether the substring is surrounded by non-alphabet
+
+    e.g.
+    ("anak anak", "anak") -> true
+    ("melaksanakan", "anak") -> false
+    ("kami harus melaksanakan", "anak") -> false
+    '''
+    if start_index >= 0:
+        end_index = start_index + len(substring)
+        return ((start_index == 0 or not string[start_index-1].isalpha()) 
+            and (end_index == len(string) or not string[end_index].isalpha()))
+    
+    return False
```

Comment:
Add a test to parser_test.py? You can just copy paste the examples you list in the documentation. Also maybe add a case w/ f("anak-anak", "anak") ?

Reply:
Btw for this, I think it's better if anak-anak, anak is defaulted to false. Unless we want to account for edge cases of repeated words. My thought is that, sometimes repeated words have a different meaning than its non-repeated counterpart